sway-core/src/asm_lang/virtual_register.rs

```diff
@@ -76,6 +76,32 @@ pub enum ConstantRegister {
     FuncArg5,
 }
 
+impl ConstantRegister {
+    pub(crate) fn parse_register_name(raw: &str) -> Option<ConstantRegister> {
```

Comment:
nit: Maybe look into implementing the `impl FromStr` trait instead.
This would introduce the `parse` function on the enum - which probably is the more idiomatic rust approach in this case.
It does however return a result instead (which I guess can be converted to an option using `.ok` or something)

Reply:
If it is no a problem I will put this into another PR.